geode-core/src/main/java/org/apache/geode/distributed/internal/locks/DLockGrantor.java

```diff
@@ -498,7 +504,13 @@ void handleLockBatch(DLockRequestMessage request) throws InterruptedException {
         }
 
         DLockBatch batch = (DLockBatch) request.getObjectName();
-        this.resMgr.makeReservation((IdentityArrayList) batch.getReqs());
+        synchronized (membersDepartedTime) {
+          // the transaction host/txLock requester has departed.
+          if (membersDepartedTime.containsKey(batch.getOwner())) {
+            throw new TransactionDataNodeHasDepartedException("Transaction host has been departed");
+          }
+          resMgr.makeReservation((IdentityArrayList) batch.getReqs());
```

Comment:
The reason the make Reservation needs to be in the sync block is to prevent this bug:
if makeReservation thread passed the sync check, and before the lock is being hold in makeReservation, there is a possibility that the memberDeparted event occurs and release all existing dlockbatch from the departed member. Otherwise the lock will be held and no threads to clean it up.

Reply:
Actually I do see the issue in my fix. I will change the order so that batchLocks will be used to sync the member departed event.